Implement MeshcatVisualizer in visualize component

GiulioRomualdi opened the pull request:
Thanks to this PR it will be possible to use meshcat-cpp in idyntree. Python and matlab bindings are provided 

The original code was taken and modified from https://github.com/GiulioRomualdi/idyntree-meshcat-cpp

**⚠️ Warning** Only meshes are supported and the color is taken from the idyntree model

traversaro commented on src/visualization/src/MeshcatVisualizer.cpp:
```diff
@@ -0,0 +1,256 @@
+        return format == "dae" || format == "stl" || format == "obj";
+    }
+
+    std::string getAbsoluteMeshPath(const std::string &modelName, const std::string &linkName, const iDynTree::ExternalMesh &externalMesh)
```
Can you add a comment to clarify what this function is doing? From the name I tought it was returning a path in the filesystem, while by reading it I guess it is returning a path in the meshcat tree.

GiulioRomualdi commented:
Hi @traversaro I removed the bindings compilation, I improved the documentation as you suggested in https://github.com/robotology/idyntree/pull/1074#discussion_r1193620706 and I implemented the span/matrixview version of the `setModelState`

traversaro commented:
> Hi @traversaro I removed the bindings compilation, I improved the documentation as you suggested in [#1074 (comment)](https://github.com/robotology/idyntree/pull/1074#discussion_r1193620706) and I implemented the span/matrixview version of the `setModelState`

Thanks! I updated the documentation and PR to contain the same information provided in the PR.
